### PR TITLE
Set idempotent attribute at form level

### PR DESCRIPTION
**Pull request by claudep**
This small refactoring is helping separate core and plugin concerns as idempotence can be defined on the forms themselves.

This is also a preparation for the future email plugin.

### CODE REVIEW

**Review by moggers87 — COMMENTED** [review]
Looks good, just not sure we have enough testing around idempotent-ness

File: two_factor/views/utils.py
Comment (claudep) [reply two_factor/views/utils.py:88]: I just added specific `is_step_visible` tests.

**Review by moggers87 — APPROVED** [review]
It appears the lines you've changed in those places didn't have test coverage, so that's why checks are failing.

I don't think that's enough to block merging this PR, but if you have the time to add some more tests to give those lines better coverage feel free to do so before merging.

**Comment by claudep** [comment]
Sorry, but I tend not to trust codecov failures much. The problem is that it fails badly as soon as you are removing lines in files that are not 100% covered. Moreover, I'm now getting `Github API rate limit error` that prevents me looking at reports. Thanks for reviewing!

**Pull request merged**
